core/src/main/java/org/apache/iceberg/FastAppend.java

```diff
@@ -114,6 +114,21 @@ public FastAppend appendManifest(ManifestFile manifest) {
     return this;
   }
 
+  @Override
+  public FastAppend toBranch(String branch) {
+    Preconditions.checkArgument(branch != null, "branch cannot be null");
+    if (ops.current().ref(branch) == null) {
+      super.createNewRef(branch);
+    }
+
+    Preconditions.checkArgument(ops.current()
+            .ref(branch).type()
+            .equals(SnapshotRefType.BRANCH),
+        "%s is not a ref to type branch", branch);
```

Comment:
Also now that we're creating a branch on commit, we should basically do a null check. It's okay if the branch is null or if it's not null, it must be a ref of type branch.